improver/cli/recursive_filter.py

```diff
@@ -34,7 +34,8 @@
 from improver import cli
 
 input_smoothing_coefficients = cli.create_constrained_inputcubelist_converter(
-    "smoothing_coefficient_x", "smoothing_coefficient_y"
+    lambda cube: cube.name() in ["smoothing_coefficient_x"],
+    lambda cube: cube.name() in ["smoothing_coefficient_y"],
```

Comment:
Not needed, `create_constrained_inputcubelist_converter` works with strings as constraints too.